qiskit_experiments/library/stark/ramsey_xy.py

```diff
@@ -0,0 +1,319 @@
+        stark_amp = np.abs(opt.stark_amp)
+        stark_channel = opt.stark_channel or pulse.DriveChannel(self.physical_qubits[0])
+        ramps_dt = self._timing.round_pulse(time=2 * opt.stark_risefall * opt.stark_sigma)
+        sigma_dt = ramps_dt / 2 / opt.stark_risefall
+
+        with pulse.build() as stark_v_schedule:
+            pulse.set_frequency(stark_freq, stark_channel)
+            pulse.play(
+                pulse.Gaussian(
+                    duration=ramps_dt,
+                    amp=stark_amp,
+                    sigma=sigma_dt,
+                ),
+                stark_channel,
+            )
+
+        with pulse.build() as stark_u_schedule:
+            pulse.set_frequency(stark_freq, stark_channel)
+            pulse.play(
+                pulse.GaussianSquare(
+                    duration=ramps_dt + param,
+                    amp=stark_amp,
+                    sigma=sigma_dt,
+                    risefall_sigma_ratio=opt.stark_risefall,
+                ),
+                stark_channel,
+            )
+
+        ram_x = circuit.QuantumCircuit(1, 1)
+        ram_x.sx(0)
+        ram_x.append(stark_v, [0])
+        ram_x.x(0)
+        ram_x.append(stark_u, [0])
+        ram_x.rz(-np.pi, 0)
+        ram_x.sx(0)
+        ram_x.measure(0, 0)
+        ram_x.metadata = {"series": "X"}
+        ram_x.add_calibration(
+            gate=stark_v,
+            qubits=self.physical_qubits,
+            schedule=stark_v_schedule,
+        )
+        ram_x.add_calibration(
+            gate=stark_u,
+            qubits=self.physical_qubits,
+            schedule=stark_u_schedule,
+        )
+
+        ram_y = circuit.QuantumCircuit(1, 1)
+        ram_y.sx(0)
+        ram_y.append(stark_v, [0])
+        ram_y.x(0)
+        ram_y.append(stark_u, [0])
+        ram_y.rz(-np.pi * 3 / 2, 0)
+        ram_y.sx(0)
+        ram_y.measure(0, 0)
+        ram_y.metadata = {"series": "Y"}
+        ram_y.add_calibration(
+            gate=stark_v,
+            qubits=self.physical_qubits,
+            schedule=stark_v_schedule,
+        )
+        ram_y.add_calibration(
+            gate=stark_u,
+            qubits=self.physical_qubits,
+            schedule=stark_u_schedule,
+        )
+
+        return ram_x, ram_y
+
+    def circuits(self) -> List[circuit.QuantumCircuit]:
+        """Create circuits.
+
+        Returns:
+            A list of circuits with a variable delay.
+        """
+        ramx_circ, ramy_circ = self.parameterized_circuits()
+        param = next(iter(ramx_circ.parameters))
+
+        circs = []
+        dt = self._backend_data.dt
+        granularity = self._backend_data.granularity
+        for delay in self.delays():
+            # Not using pulse_round method of the BackendTiming class
+            # because this method considers the minimum pulse duration.
+            # Valid delay here corresponds to the flat-top length and thus can be zero at minimum.
```

Comment:
Yes, this is another option, but more strictly this is `pulse_round(ramps_dt * dt + delay) - ramps_dt` which is bit complicated. Alternatively we can add a flag to pulse_round that controls minimum delay.

Reply:
I don't feel strongly. Adding an option to `round_pulse` to ignore the minimum delay seems okay to me. I think it is nice to use `round_pulse` when we can since it allows the details of the timing constraints to change without affecting the experiment code.

Reply:
Fair enough. Done in [67ca32d](https://github.com/Qiskit/qiskit-experiments/pull/1008/commits/67ca32de0f1a69f80cac233ddee01d41f391b6a2). Instead of adding constraint argument to every method, I added them to the constructor so that we can avoid writing tons of conditional branch inside these methods.